File: 1768. Merge Strings Alternately/mergeAlternately.py
class Solution:
  def mergeAlternately(self, word1: str, word2: str) -> str:
    merged = ''
    l = max(len(word1), len(word2))
    for i in range(l):
      if i < len(word1):
        merged += word1[i]
      if i < len(word2):
        merged += word2[i]

    return merged


word1 = "abc"
word2 = "pqr"
solution = Solution()
print('mergeAlternately: ', solution.mergeAlternately(word1, word2)) # "apbqcr"

File: 1768. Merge Strings Alternately/mergeAlternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
const mergeAlternately = (word1, word2) => {
  const merged = '';
  const len = Math.max(word1.length, word2.length);

  for (let i = 0; i < len; i++) {
    if (word1[i]) {
      merged +=word1[i];
    }

    if (word2[i]) {
      merged += word2[i];
    }
  }

  return merged;
};


const word1 = "abc";
const word2 = "pqr";
console.log('mergeAlternately: ', mergeAlternately(word1, word2)); // "apbqcr"
